### DISCUSSION TITLE
If I rename my exe, will upload work?

### DISCUSSION THREAD

**Discussion by bzuillsmith** [velopack/velopack.docs#10] [Q&A] [answered]
We rename each exe to include the version number so we are not overwriting past exe files. Will `vpk upload` know to upload this file if we rename it?

**Comment by caesay** [comment, marked as answer]
The current implementation of the upload command looks for a `assets.{channel}.json` file in the build folder for a list of files to upload. This file is generated/overwritten on every `vpk pack` so that `vpk upload` will only upload files from the most recent pack command. If you re-name your exe , you will have to also update the `assets.{channel}.json` file. 

This implementation may change in the future.
